protocol-designer/src/step-generation/utils/removePairs.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+// Skip all pairs for which `excludeWhen` returns true
+export function removePairs<T>(
+  input: Array<T>,
+  excludeWhen: (T, T) => boolean
+): Array<T> {
+  const WIDTH = 2
+  let currentPos = 0
+  const res: Array<T> = []
+  while (currentPos + 1 < input.length) {
```

Comment:
given my next suggestion (not doing the last check after the `while` loop), we'll have to account for the case where the length is only `1`
```suggestion
  while (currentPos < input.length) {
```